models/Ion_auth_model.php

```diff
@@ -268,15 +259,26 @@ public function __construct()
 	 */
 	public function hash_password($password, $salt = FALSE, $use_sha1_override = FALSE)
 	{
-		if (empty($password))
+		// Check for empty password, or password containing null char
+		if (empty($password) || strpos($password, "\0") !== FALSE)
 		{
 			return FALSE;
 		}
 
 		// bcrypt
-		if ($use_sha1_override === FALSE && $this->hash_method == 'bcrypt')
+		if ($use_sha1_override === FALSE)
 		{
-			return $this->bcrypt->hash($password);
+			// Get cost
+			if ($this->random_rounds)
```

Comment:
Maybe we should just remove the random rounds?  After a certain point it's really not adding much to the security when using bcrypt...

Reply:
Agreed, let's simplify it 😃